### PR TITLE
fix: momentum reset when jumping

### PR DESCRIPTION
**Pull request by SSelinn**
## Description

Bug: after each jump/bunny hop you lose all momentum. You didn't gain any speed. Now, with the player having 1 collider (not 3), fixing the jump bool, and tweaking the physics material on the player, it should run more smoothly and bunny hop without jittering.

## Testability

In the scene "FixMomentum" you can run around and feel if the player loses momentum after each bunny hop. Yes, you do slide a little bit if you have enough momentum.

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch
- [x] Added a scene to the game for your changes

### CODE REVIEW

File: aplib.net-demo/Assets/Scripts/Player/Movement.cs [hunk 1]
```diff
@@ -93,10 +93,11 @@ private void MovePlayer()
             {
                 _rigidbody.AddForce(_maxSpeed * _acceleration * Time.fixedDeltaTime * _horizontalVelocity.normalized);
             }
+            
         }
         else
         {
-            _rigidbody.drag = 0;
+            _rigidbody.drag = 0.2f;
```
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/Player/Movement.cs:100]: If you're assigning this a value with meaning, you should define this somewhere else. I'd add an _airDrag field

File: aplib.net-demo/Assets/Scripts/Player/Movement.cs [hunk 2]
```diff
@@ -133,11 +134,10 @@ private void LimitVelocity(bool isOnWalkableSlope)
...
         else
         {
-            Vector3 rigidbodyHorizontalVelocity = new(_rigidbody.velocity.x, 0, _rigidbody.velocity.z);
-            if (rigidbodyHorizontalVelocity.magnitude > _maxSpeed)
+            if (_rigidbody.velocity.magnitude > _maxSpeed && _isGrounded)
             {
-                Vector3 limitedVelocity = rigidbodyHorizontalVelocity.normalized * _maxSpeed;
-                _rigidbody.velocity = new Vector3(limitedVelocity.x, _rigidbody.velocity.y, limitedVelocity.z);
+
+                _rigidbody.velocity = _rigidbody.velocity * 0.95f;
```
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/Player/Movement.cs:140]: You should also define this 0.95f somewhere.
Also, please update the method summary to align with your changes.

File: aplib.net-demo/Assets/Scripts/Player/Movement.cs [hunk 3]
```diff
@@ -208,5 +206,7 @@ private void HandleJump()
     /// Set the jumpPressed flag to true when the player presses the jump button.
     /// Result of an InputAction.
     /// </summary>
-    public void OnJumpPressed() => _jumpPressed = true;
+    public void OnJumpDown() => _jumpPressed = true;
+
```
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/Player/Movement.cs:210]: You should add a Method summary here

File: aplib.net-demo/Assets/Scripts/Player/Movement.cs
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/Player/Movement.cs:1]: Please run dotnet format, you have some empty lines in certain places :)

**Review by JoenvandeVorle — CHANGES_REQUESTED** [review]

File: aplib.net-demo/Assets/Materials/ColorTexture.meta
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Materials/ColorTexture.meta:1]: I'd leave the folder structure here as it was. This was also already added in another PR

File: aplib.net-demo/Assets/Scripts/Player/Movement.cs [hunk 4]
```diff
@@ -117,32 +119,34 @@ private void HandleWallCollision(Vector3 wallNormal)
...
     /// </summary>
     /// <param name="isOnWalkableSlope">Is the player on a walkable slope</param>
-    private void LimitVelocity(bool isOnWalkableSlope)
+    private void LimitVelocity(bool isOnSlope)
     {
-        if (isOnWalkableSlope)
+        if (isOnSlope)
         {
             if (_rigidbody.velocity.magnitude > _maxSpeed) _rigidbody.velocity = _rigidbody.velocity.normalized * _maxSpeed;
         }
```
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/Player/Movement.cs:134]:
```suggestion
        if (isOnSlope && _rigidbody.velocity.magnitude > _maxSpeed)
        {
            _rigidbody.velocity = _rigidbody.velocity.normalized * _maxSpeed;
            return;
        }
```

File: aplib.net-demo/Assets/Scripts/Player/Movement.cs [hunk 5]
```diff
@@ -117,32 +119,34 @@ private void HandleWallCollision(Vector3 wallNormal)
...
-            Vector3 rigidbodyHorizontalVelocity = new(_rigidbody.velocity.x, 0, _rigidbody.velocity.z);
-            if (rigidbodyHorizontalVelocity.magnitude > _maxSpeed)
-            {
-                Vector3 limitedVelocity = rigidbodyHorizontalVelocity.normalized * _maxSpeed;
-                _rigidbody.velocity = new Vector3(limitedVelocity.x, _rigidbody.velocity.y, limitedVelocity.z);
-            }
+            Vector3 horizontalVelocityLimited = rigidbodyHorizontalVelocity.normalized * _maxSpeed;
+            _rigidbody.velocity = new Vector3(horizontalVelocityLimited.x, _rigidbody.velocity.y, horizontalVelocityLimited.z);
+        }
+        if (rigidbodyVerticalVelocity.magnitude > _maxSpeed)
```
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/Player/Movement.cs:144]:
```suggestion
        }
        
        if (rigidbodyVerticalVelocity.magnitude > _maxSpeed)
```

File: aplib.net-demo/Assets/Materials/PlayerPhysic.physicMaterial [hunk 6]
```diff
@@ -7,8 +7,8 @@ PhysicMaterial:
   m_PrefabInstance: {fileID: 0}
   m_PrefabAsset: {fileID: 0}
   m_Name: PlayerPhysic
-  dynamicFriction: 0.6
-  staticFriction: 0.1
+  dynamicFriction: 0.5
```
Comment (JensSteenmetz) [inline aplib.net-demo/Assets/Materials/PlayerPhysic.physicMaterial]: Friction 0.2 to 0.3 feels better imo.
```suggestion
  dynamicFriction: 0.25
```

**Review by JensSteenmetz — APPROVED** [review]
LGTM!

**Pull request merged**